### PR TITLE
[WFGP-224] Separating code needed to run forked WfConfigGenerator

### PR DESCRIPTION
**Pull request by spyrkob**
Issue: https://issues.redhat.com/browse/WFGP-224
@jfdenise could you review? I tried to remove galleon core classes from code run in forked mode. Alternatively maybe it would be possible to change the class loader used by the forked mode.

### CODE REVIEW

File: galleon-plugins/src/main/java/org/wildfly/galleon/plugin/server/ForkedProcessRunner.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Properties;
+
+import org.jboss.galleon.Errors;
```
Comment (jfdenise) [inline galleon-plugins/src/main/java/org/wildfly/galleon/plugin/server/ForkedProcessRunner.java:29]: @spyrkob , ForkedProcessRunner class is expected to be run in a context where galleon core classes are not present. This import seems un-wanted.

Comment (spyrkob) [reply galleon-plugins/src/main/java/org/wildfly/galleon/plugin/server/ForkedProcessRunner.java]: Yes, sorry I missed it. Removed it now

File: config-gen/src/main/java/org/wildfly/galleon/plugin/config/generator/ForkedConfigGenerator.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+import java.util.Arrays;
+
+import org.jboss.dmr.ModelNode;
+import org.wildfly.galleon.plugin.WfConstants;
+import org.wildfly.galleon.plugin.server.ForkCallback;
+import org.wildfly.galleon.plugin.server.ConfigGeneratorException;
+
+public class ForkedConfigGenerator extends BaseConfigGenerator implements ForkCallback {
+
+   private final ForkCallback callback;
```
Comment (jfdenise) [inline config-gen/src/main/java/org/wildfly/galleon/plugin/config/generator/ForkedConfigGenerator.java:34]: This field seems un-used.

File: config-gen/src/main/java/org/wildfly/galleon/plugin/config/generator/ForkedConfigGenerator.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+   @Override
+   public void forkedForEmbedded(String... args) throws ConfigGeneratorException {
+      if(args.length != 2) {
+         throw new IllegalArgumentException("Expected one argument but received " + Arrays.asList(args));
+      }
+      this.jbossHome = args[0];
+      final Path script = Paths.get(args[1]);
+      if(!Files.exists(script)) {
+         throw new ConfigGeneratorException("Errors.pathDoesNotExist(script)");
```
Comment (jfdenise) [inline config-gen/src/main/java/org/wildfly/galleon/plugin/config/generator/ForkedConfigGenerator.java:48]: The message doesn't seem correct.

File: config-gen/src/main/java/org/wildfly/galleon/plugin/config/generator/WfConfigsReader.java [hunk 4]
```diff
@@ -354,8 +362,8 @@ public void forkedEmbeddedDone(String... args) throws ProvisioningException {
                 for(Path xml : stream) {
                     addedConfigs = CollectionUtils.add(addedConfigs, ProvisionedConfigXmlParser.parse(xml));
                 }
-            } catch (IOException e) {
-                throw new ProvisioningException(Errors.readDirectory(p), e);
+            } catch (Exception e) {
```
Comment (jfdenise) [inline config-gen/src/main/java/org/wildfly/galleon/plugin/config/generator/WfConfigsReader.java:365]: Could this exception be IOException?

File: config-gen/src/main/java/org/wildfly/galleon/plugin/config/generator/WfConfigsReader.java [hunk 5]
```diff
@@ -364,8 +372,8 @@ public void forkedEmbeddedDone(String... args) throws ProvisioningException {
                 for(Path xml : stream) {
                     updatedConfigs = CollectionUtils.add(updatedConfigs, ProvisionedConfigXmlParser.parse(xml));
                 }
-            } catch (IOException e) {
-                throw new ProvisioningException(Errors.readDirectory(p), e);
+            } catch (Exception e) {
```
Comment (jfdenise) [inline config-gen/src/main/java/org/wildfly/galleon/plugin/config/generator/WfConfigsReader.java:375]: Could this exception be IOException?

File: config-gen/src/main/java/org/wildfly/galleon/plugin/config/generator/ForkedConfigGenerator.java [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+      }
+      try {
+         executeScript(script);
+      } catch(Exception e) {
```
Comment (jfdenise) [inline config-gen/src/main/java/org/wildfly/galleon/plugin/config/generator/ForkedConfigGenerator.java:52]: Only IOException should be caught, a ConfigGeneratorException could have been raised.

**Comment by jfdenise** [comment]
@spyrkob , could you change the commit comment, I moved the issue to galleon-plugins: WFGP-224 Thank-you.

**Comment by spyrkob** [comment]
@jfdenise I updated the commit and exception handling

**Comment by jfdenise** [comment]
Merged @spyrkob , thank-you!

**Pull request merged**